pip/index.py

```diff
@@ -640,6 +642,12 @@ def _link_package_versions(self, link, search):
                 self._log_skipped_link(
                     link, 'Python version is incorrect')
                 return
+
+        if not check_requires_python(link.requires_python):
```

Comment:
I'd say we certainly don't want to raise and crash `pip install` because an index server has a malformed `data-requires`.
In such case, we can either ignore the malformed `data-requires-python` (and accept the link) or decide to ignore the link altogether.